lib/client/renderer.js

```diff
@@ -418,26 +425,110 @@ function init (client, d3) {
   }
 
   function appendTreatments(treatment, arc) {
+    
+    function boluscalcTooltip (treatment) {
+      if (!treatment.boluscalc) {
+        return '';
+      }
+      var html = '<hr>';
+      html += (treatment.boluscalc.othercorrection ? '<strong>'+translate('Other correction')+':</strong> ' + parseFloat(treatment.boluscalc.othercorrection).toFixed(2) + 'U<br/>' : '');
+      html += (treatment.boluscalc.profile ? '<strong>'+translate('Profile used')+':</strong> ' + treatment.boluscalc.profile + '<br/>' : '');
+      if (treatment.boluscalc.foods && treatment.boluscalc.foods.length) {
+        html += '<table><tr><td><strong>' + translate('Food') + '</strong></td></tr>';
+        for (var fi=0; fi<treatment.boluscalc.foods.length; fi++) {
+          var f = treatment.boluscalc.foods[fi];
+          html += '<tr>';
+          html += '<td>'+ f.name + '</td>';
+          html += '<td>'+ (f.portion*f.portions).toFixed(1) + ' ' + f.unit + '</td>';
+          html += '<td>('+ (f.carbs*f.portions).toFixed(1) + ' g)</td>';
+          html += '</tr>';
+        }
+        html += '</table>';
+      }
+      return html;
+    }
+    
     function treatmentTooltip() {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2002418912/issues/source?bid=2452374&fileBranchId=3122262#l451)